ogr/abstract.py

```diff
@@ -404,6 +404,14 @@ def patch(self) -> bytes:
     def head_commit(self) -> str:
         raise NotImplementedError
 
+    @property
+    def merge_commit_sha(self) -> str:
+        raise NotImplementedError
```

Comment:
```suggestion
        raise NotImplementedError()
```

an instance should be raised instead of the class

Reply:
It does not matter, you can use class or and instance: https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement

>  raise evaluates the first expression as the exception object. It must be either a subclass or an instance of BaseException. If it is a class, the exception instance will be obtained when needed by instantiating the class with no arguments.

(https://stackoverflow.com/questions/16706956/is-there-a-difference-between-raise-exception-and-raise-exception-without#16709222)

I know it works but this was the first time I've checked it in the docs..;) So, thanks for pointing that out!

But it would be nice to unify this across the project.

Reply:
As I mentioned on IRC, in GitHub, 'mergeable' is a boolean, while in
GitLab, the closest thing seems to be merge_status which, if I'm
reading the doc (https://docs.gitlab.com/ee/api/merge_requests.html)
correctly, is a string that can have several values including "unchecked",
"checking", "can_be_merged", and "cannot_be_merged".

I went on to say that "...Maybe the answer is to map GL's
"can_be_merged" to 'True' and everything else to 'False'" and Tomas
like this idea, at least provisionally.

OR we could map "can_be_merged" to True, "cannot_be_merged" to False,
and everything else to None.

Reply:
In the Packit Arthitecture meeting on 06/15/2021, we decided that we
would present GitLab-like behavior, and map GitHub behavior onto it.

This means that we will provide a (GitLab-like) 'merge_status'
@property instead of (GitHub-like) 'mergeable', with GitHub's
'mergeable' values mapped as follows:

mergeable    => merge_status
True            "can_be_merged"
False           "cannot_be_merged"
-               "unchecked"
-               "checking"
-               "cannot_be_merged_recheck"